### DISCUSSION TITLE
posts/tls-cryptography-6-math/

### DISCUSSION THREAD

**Discussion by giscus** [humminglab/blog.humminglab.io#12] [Announcements]
# posts/tls-cryptography-6-math/

키 합의, 비대칭키는 역연산이 실제적으로 불가능한 수학적 이론을 기반으로 만들어진 알고리즘이기 때문에, 이의 기본 동작을 이해하기 위해서는 수학적인 지식이 어느정도 필요하다.
여기에서는 이들 알고리즘의 동작 원리에 필요한 최소한의 수학을 정리해본다.
타원알고리즘을 제외하고, 키합의와 비대칭키 알고리즘을 이해하기 위해서는 다음과 같은 것을 알아야 한다.
소인수 분해 유클리드 호제법을 이용하여 두 수의 최대 공약 수 찾기 확장 유클리드 호제법으로 소수 모듈로 연산에서 곱하기 역원 찾기 페르마 소정리를 이용하여 소수 모듈로의 지수 연산 역원 찾기 오일러 정리를 이용하여 일반 모듈로에서 지수 역원 찾기 소인수 분해임의의 정수를 소인수 분해하는 데에는 효과적인 방법이 없다.

https://blog.humminglab.io/posts/tls-cryptography-6-math/

**Comment by Morroh22** [comment]
안녕하세요? 정리해 놓으신 자료를 잘 보고 있습니다.
쉽게 정리해주셔서 감사 말씀올립니다.
다만 '페르마의 작은 정리'에 표현해 놓은 공식이 오류가 있는 것 같아 한 글자 적습니다. 
$ k^{p-1} \equiv 1 \mod p$ 로 표현된 것이
$ k^{p-1} \equiv k \mod p$ 로 표현되어야 할 것 같습니다.

**Reply by aqwerf** [reply]
커멘트 남겨주셔서 감사합니다.

제가 정리해 놓고도 가물가물 해서 다시 읽어 보았는데, 해당 부분은 아래와 같은 수식이 맞습니다. 

$ k^{p-1} \equiv 1 \mod p$

지수가 (p-1) 인 경우에 1 이 되고, p 인 경에는 원래 수 k 가 됩니다. 
혹시 제가 잘못 정리된 부분이 있다면 다시 알려주세요.

**Comment by echoja** [comment]
안녕하세요 수학 초보입니다.

> 위의 소수 3과 5의 곱인 모듈로 15를 보면 다음과 같다.
> $$ k^8 \equiv 1 \mod 15 $$
> 이때 k는 1~14까지의 어떤 수이어도 가능하다.

에서 k는 15의 서로소인건 아닌가요? 3^8 mod 15 = 6 로 나와서요...

**Reply by aqwerf** [reply]
알려주셔서 감사합니다.

네, 말씀하신 것처럼 서로소인 경우만 가능한데 잘못 적었군요.
해당 페이지를 오타와 문구도 같이 수정하였습니다.

**Reply by echoja** [reply]
양질의 글 덕분에 큰 도움이 되고 있습니다. 감사합니다.
수학 베이스가 약해서 천천히 학습하고 있네요. 끝까지 정진하겠습니다.
